### DISCUSSION TITLE
buildSearchIndexes for remote files

### DISCUSSION THREAD

**Discussion by KhasVN** [fuma-nama/fumadocs#1027] [Docs Feedback]
Hi, I was impressed by repo, I already implemented this for my docs!!

I found out that the function { buildSearchIndexes } from '@fumadocs/mdx-remote/github'; always use local files

The code of that function here:
```js 
async function buildSearchIndexes(options) {
  const {
    slug = getSlugs,
    baseUrl = "/",
    url = createGetUrl(baseUrl),
    ...rest
  } = options;
  const files = await getLocalFiles({
    ...rest,
    keepContent: true
  });
  ...
```

I followed this [Docs](https://fumadocs.vercel.app/docs/headless/remote/github#search), everything works fine but not the search function, both in production and development env, it doesn't fetch the files from remote repo. 

Will the function get updated, 1 more options field for local or remote?
Thanks so much!! Have a gooooood day!!!

**Comment by fuma-nama** [comment]
it requires reading the content of all files, which has a very bad performance for remote. You can imagine 1000+ HTTP requests being sent to GitHub API to fetch file contents, if you've lots of documents.

You're recommended to build the search indexes during a build process (e.g. triggered every commit/PR merge) and store the output somewhere, like Algolia or Orama Cloud, and use their cloud integration to implement search.

You can also store it in a public URL (e.g. in a Git repo), and use `fetch()` to fetch search indexes from that URL. Pass search indexes to [`createSearchAPI`](https://fumadocs.vercel.app/docs/headless/search/orama#from-search-indexes) and make your own route handler for search endpoint.

**Reply by KhasVN** [reply]
TYSM!! It works! Sorry for asking for a dump thing!
